### PR TITLE
feat(component-protocol): add component cardlist

### PR DESCRIPTION
**Pull request by bugaolengdeyuxiaoer**
#### What type of this PR

Add one of the following kinds:
/kind feature

#### Which issue(s) this PR fixes:
Fixes #

#### Specified Reivewers:
/assign @sfwn

### CODE REVIEW

File: providers/component-protocol/components/cardlist/interface.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// ICardList .
+type ICardList interface {
+	cptype.IComponent
+	ICardListStdOps
+}
+
+// ICardListStdOps .
+type ICardListStdOps interface {
+	RegisterCardListOp(opData OpCardListGoto) (opFunc cptype.OperationFunc)
+	RegisterCardListIconOp(opData OpCardListIconGoto) (opFunc cptype.OperationFunc)
```
Comment (sfwn) [inline providers/component-protocol/components/cardlist/interface.go:30]: Maybe RegisterCardIconGotoOp op is better?

`RegisterCardIconGotoOp(opData OpCardIconGoto)`

File: providers/component-protocol/components/cardlist/model.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+type (
+	// Data cardList std data
+	Data struct {
+		Total int
+		Title string
+		List  []List `json:"list,omitempty"`
+	}
+	// List .
+	List struct {
```
Comment (sfwn) [inline providers/component-protocol/components/cardlist/model.go:27]: Use `Card`

Comment (sfwn) [reply providers/component-protocol/components/cardlist/model.go:27]: `Card` need ID, Title and others.

File: providers/component-protocol/components/cardlist/model.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		Logo  string
+		Title string
+		Label string
+		Star  bool
+		Extra cptype.Extra
```
Comment (sfwn) [inline providers/component-protocol/components/cardlist/model.go:32]: cptype.Extra already wrapped a field called `extra`.

File: providers/component-protocol/components/cardlist/model.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+type (
+	// Data cardList std data
+	Data struct {
+		Total int
+		Title string
+		List  []List `json:"list,omitempty"`
+	}
+	// List .
+	List struct {
+		Logo  string
```
Comment (sfwn) [inline providers/component-protocol/components/cardlist/model.go:28]: use LogoURL uniformly.

File: providers/component-protocol/components/cardlist/model.go [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+		Title string
+		Label string
```
Comment (sfwn) [inline providers/component-protocol/components/cardlist/model.go:30]: What is the `label` means?

Maybe you should use `labels` and define a struct for label.

File: providers/component-protocol/components/cardlist/operation.go [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+type (
+	// OpCardListGoto .
+	OpCardListGoto struct {
+		cptype.Operation
+		ServerData OpCardListGotoData `json:"serverData,omitempty"`
+	}
+
+	// OpCardListGotoData .
+	OpCardListGotoData struct {
+		JumpOut bool
```
Comment (sfwn) [inline providers/component-protocol/components/cardlist/operation.go:30]: Add comment for user pls.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — CHANGES_REQUESTED** [review]
add json tag for models.

**Comment by bugaolengdeyuxiaoer** [comment]
> add json tag for models.

finished

**Review by sfwn — APPROVED** [review]

**Pull request merged**